### ISSUE TITLE
Fuzzy bookmark/quickmark search

### ISSUE THREAD

**Issue by adigitoleo** [qutebrowser/qutebrowser#6871]
Wishlist feature request for less strict bookmark/quickmark completion matching.

Currently, bookmark/quickmark completion uses non-commutative string matching. Consider the mockup:

```
https://example.com/PAGE/name_1.html                              Example Page
```

The following seaerch, e.g. in the `:open` prompt, would **not** bring that URL onto the suggestion list:

```
1 name
```

Some care may need to be taken to ensure this doesn't have any noticable performance impact. Also, for reference, there is `search.ignore_case` for page content searches, so maybe this could also be optional.

Might be worth waiting on the new bookmark system, see #882.

**Comment by rcorre** [comment]
Bookmark/quickmark seems case-insensitive to me
![1639242050](https://user-images.githubusercontent.com/2496231/145685082-e21a173c-cbdd-4928-a852-26ea4320c6bc.png)
![1639241984](https://user-images.githubusercontent.com/2496231/145685083-f59db3b7-4884-452e-bd51-ead8cdbc0587.png)

**Comment by rcorre** [comment]
The examples you gave work as well:

![1639242315](https://user-images.githubusercontent.com/2496231/145685225-a6f50920-38c1-4ba1-aa62-cbead2e209fa.png)

**Comment by rcorre** [comment]
Unless you are searching against some non-latin unicode characters: https://github.com/qutebrowser/qutebrowser/issues/6738#issuecomment-950311623.

**Comment by adigitoleo** [comment]
@rcorre Thanks, you're right it is in fact case insensitive. Somehow I missed that. I'll modify the OP to only mention the fuzzy searching.

**Comment by The-Compiler** [comment]
This looks like a duplicate of #6114 then.
